parser/parser.go

```diff
@@ -19,16 +19,7 @@ import (
 // ParseError (actually *ParseError) is the type of error returned by
 // ParseProgram.
 type ParseError struct {
-	// Source line/column position where the error occurred.
-	Position Position
-	// Error message.
-	Message string
-}
-
-// Error returns a formatted version of the error, including the line
-// and column numbers.
-func (e *ParseError) Error() string {
-	return fmt.Sprintf("parse error at %d:%d: %s", e.Position.Line, e.Position.Column, e.Message)
+	PositionError
```

Comment:
Because this will now be `ast.PositionError` (per comments in `lexer`), let's revert the change to these fields (mainly for the sake of API documentation), and just copy the fields over when we want to create the `ParseError`.

We'll need to add `ParseError.Error` back too.